components/eam/src/chemistry/yaml/aero_model/f90_yaml/aero_model_wetdep_beg_yml.f90

```diff
@@ -0,0 +1,188 @@
+        call write_var(unit_input,unit_output,'ncol',state%ncol)
+        call write_var(unit_input,unit_output,'state_q',state%q(yaml%col_print,yaml%lev_print,:)) ! ??? distinct from state_q used later on?
+        call write_var(unit_input,unit_output,'temperature',state%t(yaml%col_print,yaml%lev_print))
+        call write_var(unit_input,unit_output,'pmid',state%pmid(yaml%col_print,yaml%lev_print))
+        call write_var(unit_input,unit_output,'pdel',state%pdel(yaml%col_print,yaml%lev_print)) 
+
+        ! the following are input arguments that are fields within pbuf.  We thus need to duplicate the pbuf extraction subroutine calls in order to write out the fields.
+
+        call pbuf_get_field(pbuf, dgnum_idx,      dgncur_a_yaml )  ! note:  this is called again later on before passed into dgnumdry_m in modal_aero_calcsize_sub
+        call pbuf_get_field(pbuf, wetdens_ap_idx, wetdens_yaml)
+        call pbuf_get_field(pbuf, qaerwat_idx,    qaerwat_yaml)
+        call pbuf_get_field(pbuf, dgnumwet_idx, dgnumwet_yaml, start=(/1,1,1/), kount=(/pcols,pver,nmodes/) )
+        call pbuf_get_field(pbuf, fracis_idx,   fracis_yaml,   start=(/1,1,1/), kount=(/pcols,pver, pcnst/) )
+        call pbuf_get_field(pbuf, rprddp_idx,      rprddp_yaml  )
+        call pbuf_get_field(pbuf, rprdsh_idx,      rprdsh_yaml  )
+        call pbuf_get_field(pbuf, nevapr_shcu_idx, evapcsh_yaml )
+        call pbuf_get_field(pbuf, nevapr_dpcu_idx, evapcdp_yaml )
+
+        call write_var(unit_input,unit_output,'dgncur_a',dgncur_a_yaml(yaml%col_print,yaml%lev_print,:)) 
+        call write_var(unit_input,unit_output,'wetdens',wetdens_yaml(yaml%col_print,yaml%lev_print,:))  
+        call write_var(unit_input,unit_output,'qaerwat',qaerwat_yaml(yaml%col_print,yaml%lev_print,:)) 
+        call write_var(unit_input,unit_output,'dgnumwet',dgnumwet_yaml(yaml%col_print,yaml%lev_print,:))  
+        call write_var(unit_input,unit_output,'fracis',fracis_yaml(yaml%col_print,yaml%lev_print,:)) 
+        call write_var(unit_input,unit_output,'rprddp',rprddp_yaml(:,:)) 
+        call write_var(unit_input,unit_output,'rprdsh',rprdsh_yaml(:,:)) 
+        call write_var(unit_input,unit_output,'evapcsh',evapcsh_yaml(:,:)) 
+        call write_var(unit_input,unit_output,'evapcdp',evapcdp_yaml(:,:)) 
+
+        call get_cldbrn_mmr(lchnk, pbuf, &! in
+         qqcw_yaml) !out      
+        
+        ! write out qqcw fields only over constituents that are set, for others set value to -9999.9
+  
+        if (allocated(qqcw_yaml_2darr)) deallocate(qqcw_yaml_2darr)
+        allocate(qqcw_yaml_2darr(size(qqcw_yaml),pver))
+        do imm=1,size(qqcw_yaml)
+           if (imm<=15) then
+                qqcw_yaml_2darr(imm,:) = -9999.9
+           else
+                qqcw_yaml_2darr(imm,:) = qqcw_yaml(imm)%fld(yaml%col_print,:)
+           endif
+        enddo
+        call write_var(unit_input,unit_output,'qqcw',qqcw_yaml_2darr)
+        deallocate(qqcw_yaml_2darr)
+
```

Comment:
Since ptend is a data structure, normally we don't write it out, just individual components of ptend.  In any event, ptend is intent-out for this subroutine, so it should not be in the input files; I have tried to put all the fields within ptend generated on output in the output .py files.

Reply:
We only need tend%q.